game-core/src/main/java/games/strategy/engine/chat/ChatMessageListener.java

```diff
@@ -6,7 +6,7 @@
 public interface ChatMessageListener {
   void addMessage(String message, PlayerName from);
 
-  void addMessageWithSound(String message, PlayerName from, String sound);
+  void addSlapMessage(String message, PlayerName from);
 
   void addStatusMessage(String message);
```

Comment:
This is a thing I'm not quite sure about:
All those kind of messages are basically the same, so I was thinking maybe it makes sense to merge all those methods and add a `MessageType` enum.
This would make sense because 2 out of 3 implementations have the same implementation for `addMessage` and `addSlapMessage`, so they call the other method anyways.
However if we did this for `addSlapMessage` and `addMessage`, we'd have to do this for `addStatusMessage` as well if we want to stay consistent. But then we'd have to pass null for `PlayerName from` because this information is not present in this case.
That's why I decided to do it this way for now. Let me know if you think the other approach is better in your opinion

Reply:
I think this is okay for now. In part the question is do we have the control flow be at the caller or callee, ie: this is a chat message, this is a player joined message. I think keeping strong types and enumerating the different event types on this interface is good for now. This way the listener can be a true listener and it's caller will identify and then route messages to it.